File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/SummaryValues/CustomGanttChartItem.cs
using DlhSoft.Windows.Controls;
using System.ComponentModel;
using System.Linq;

namespace Demos.WPF.CSharp.GanttChartDataGrid.SummaryValues
{
    class CustomGanttChartItem : GanttChartItem, INotifyPropertyChanged
    {
        private decimal extraCosts;
        public decimal ExtraCosts
        {
            get { return extraCosts; }
            set
            {
                if (value < 0)
                    value = 0;

                extraCosts = value;
                OnPropertyChanged("ExtraCosts");

                if (GanttChartView == null)
                    return;
                CustomGanttChartItem parent = GanttChartView.GetParent(this) as CustomGanttChartItem;
                if (parent == null)
                    return;
                parent.ExtraCosts = GanttChartView.GetChildren(parent).Sum(item => (item as CustomGanttChartItem).ExtraCosts);
            }
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-VisualBasic/GanttChartDataGrid/SummaryValues/CustomGanttChartItem.vb
Imports DlhSoft.Windows.Controls
Imports System.ComponentModel

Friend Class CustomGanttChartItem
    Inherits GanttChartItem
    Implements INotifyPropertyChanged

    Private extraCostsValue As Decimal
    Public Property ExtraCosts() As Decimal
        Get
            Return extraCostsValue
        End Get
        Set(value As Decimal)
            If value < 0 Then
                value = 0
            End If

            extraCostsValue = value
            OnPropertyChanged("ExtraCosts")

            If GanttChartView Is Nothing Then
                Return
            End If
            Dim parent As CustomGanttChartItem = TryCast(GanttChartView.GetParent(Me), CustomGanttChartItem)
            If parent Is Nothing Then
                Return
            End If
            parent.ExtraCosts = GanttChartView.GetChildren(parent).Sum(Function(item) (TryCast(item, CustomGanttChartItem)).ExtraCosts)
        End Set
    End Property
End Class
